marimo/_runtime/executor.py

```diff
@@ -95,6 +95,37 @@ async def execute_cell_async(
         pass
 
 
+def handle_execute_cell_exception(
+    e: Exception, cell: CellImpl, glbls: dict[str, Any]
+) -> None:
+    tb = e.__traceback__
+    while tb and tb.tb_next:
+        tb = tb.tb_next
+
+    if tb is None:
+        raise e
+
+    filename = f"__marimo__cell_{cell.cell_id}_.py"
+
+    if filename in tb.tb_frame.f_code.co_filename and cell._start_line:
```

Comment:
Is this satisfied only for executing as a script, because `cell._start_line` is nonnegative only in that case?